devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,223 @@
+            add_sycl=True,
+        )
+        download(
+            self.directory,
+            self.grappa_url(),
+            self.directory / self.grappa_file(),
+            checksum="cc02be35ba85c8b044e47d097661dffa8bea57cdb3db8b5da5d01cdbc94fe6c8902652cfe05fb9da7f2af0698be283a2",
+            untar=True,
+        )
+
+    def teardown(self):
+        pass
+
+
+class GromacsBenchmark(Benchmark):
+    def __init__(self, suite, model, type, option):
+        self.suite = suite
+        self.model = model  # The model name (e.g., "0001.5")
+        self.type = type
+        self.gromacs_src = suite.gromacs_src
+        self.grappa_dir = suite.grappa_dir
+        self.gmx_path = suite.gromacs_build_path / "bin" / "gmx"
+        self.option = option
+
+        if self.type == "pme":
+            self.extra_args = [
+                "-pme",
+                "gpu",
+                "-pmefft",
+                "gpu",
+                "-notunepme",
+            ]
+        else:
+            self.extra_args = []
+
+    def name(self):
+        return f"gromacs-{self.model}-{self.type}-{self.option}"
+
+    def setup(self):
+        if self.type != "rf" and self.type != "pme":
+            raise ValueError(f"Unknown benchmark type: {self.type}")
+
+        if self.option != "graphs" and self.option != "eager":
+            raise ValueError(f"Unknown option: {self.option}")
+
+        if not self.gmx_path.exists():
+            raise FileNotFoundError(f"gmx executable not found at {self.gmx_path}")
+
+        model_dir = self.grappa_dir / self.model
+
+        if not model_dir.exists():
+            raise FileNotFoundError(f"Model directory not found: {model_dir}")
+
+        cmd_list = [
+            str(self.gmx_path),
+            "grompp",
+            "-f",
+            f"{str(self.grappa_dir)}/{self.type}.mdp",
+            "-c",
+            str(model_dir / "conf.gro"),
+            "-p",
+            str(model_dir / "topol.top"),
+            "-o",
+            f"{str(model_dir)}/{self.type}.tpr",
+        ]
+
+        # Generate configuration files
+        self.conf_result = run(
+            cmd_list,
+            add_sycl=True,
+        )
+
+    def run(self, env_vars):
+        model_dir = self.grappa_dir / self.model
+
+        env_vars.update({"SYCL_CACHE_PERSISTENT": "1"})
+
+        if self.option == "graphs":
+            env_vars.update({"GMX_CUDA_GRAPH": "1"})
+
+        # Run benchmark
+        command = [
+            str(self.gmx_path),
+            "mdrun",
+            "-s",
+            f"{str(model_dir)}/{self.type}.tpr",
+            "-nb",
+            "gpu",
+            "-update",
+            "gpu",
+            "-bonded",
+            "gpu",
+            "-ntmpi",
+            "1",
+            "-ntomp",
+            "1",
+            "-nobackup",
+            "-noconfout",
+            "-nstlist",
+            "100",
+            "-pin",
+            "on",
```

Comment:
This will discard the first half of the run from the reported performance, which should prevent any effects from, e.g., initial JITting of the kernels.

```suggestion
            "on",
            "-resethway",
```

Reply:
Done